flake.nix

```diff
@@ -19,7 +20,21 @@
           # instead `hpkgs.developPackage` compiles Setup.hs with ghc
           # then runs ./Setup several times. This is implemented at
           # https://github.com/NixOS/nixpkgs/blob/a781ff33ae/pkgs/development/haskell-modules/generic-builder.nix
-          root = ./.;
+
+          # A minimal set of files to copy into nix's build sandbox.
+          # Whenever these files change, `nix build` recompiles Agda.
+          root = fs.toSource {
+            root = ./.;
+            fileset = fs.unions [
+              ./src/setup
+              ./src/full
+              ./src/main
+              ./src/data
+              ./src/agda-mode
+              ./Agda.cabal
+              ./LICENSE
+            ];
```

Comment:
Any reason for not just including all of `src`?

Reply:
`src/` also contains some non-Agda source code, e.g. for GitHub actions, and for developer tools like `agda-bisect` and version-bumping scripts.